lib/gateway/index.js

```diff
@@ -76,23 +79,37 @@ function bootstrap({ plugins, config } = {}) {
       conditionEngine.register(cond);
     });
   }
+};
+
+function bootstrap({ plugins, config } = {}) {
+  let rootRouter;
+  const app = express();
+  app.set('x-powered-by', false);
   app.use(passport.initialize());
+  bootstrapPolicies({ app, plugins, config });
   rootRouter = pipelines.bootstrap({ app: express.Router(), config });
   app.use((req, res, next) => {
-    // rootRouter will process all requests;
-    // after hot swap old instance will continue to serve previous requests
-    // new instance will be serving new requests
-    // once all old requests are served old instance is target for GC
     rootRouter(req, res, next);
   });
 
   eventBus.on('hot-reload', (hotReloadContext) => {
+    const oldConfig = config;
+    const oldPlugins = plugins;
     const oldRootRouter = rootRouter;
     try {
-      rootRouter = pipelines.bootstrap({ app: express.Router(), config: hotReloadContext.config });
-      log.info('hot-reload router completed');
+      const newConfig = hotReloadContext.config;
+      policies.removeAll();
```

Comment:
I do not think this is a good idea. What we should be doing here is to check out and load only the *new* policies instead of ditching everything and reload them from scratch. It should not be a big deal, but I believe it's important.

Reply:
This was added to cater the case when a policy is removed from the config.
If we want to selectively add or remove policies according to the new config, then I think we would have to expose two functions to check if a policy exists and other to remove the policy removed from config.

What do you think about this?

Reply:
I'd go in this direction, as long it is not that difficult. The whole idea is to minimize the startup time and the reload time, so it's kind of important to make sure we do not waste time on that.

Reply:
I agree with your point. I'm now working on to check and then load or remove core policies as well as plugins.